### DISCUSSION TITLE
assignToPanels method was removed

### DISCUSSION THREAD

**Discussion by baszach** [laravel/nova-issues#4538] [Q&A] [answered]
Hello,

I've come across a piece of legacy code of another dev using the method `assignToPanels`. This method doesn't seem to exist anymore. Here's the legacy code:

` public function updateFieldsWithinPanels(NovaRequest $request, BaseResource $resource = null) {`
        `return $this->assignToPanels(__('__SOME_TEXT__'), $this->updateFields($request));`
    `}`

Inside a Nova resource. What is this method doing and why was it removed?

Any help would be appreciated :)

**Comment by crynobone** [comment, marked as answer]
The method was refactored in Nova 4 major release, you will need to update `updateFieldsWithinPanels` to match with the new code structure.

**Reply by baszach** [reply]
I understand that the method was refactored. My issue is that I don't fully understand the structure and functionality and cannot find it in the docs. I was hoping to get some info on the `assignToPanels` method because it doesn't exist anymore.

**Reply by crynobone** [reply]
We generally don't documents change on non-public API. I also feels you doing something odd here and I don't want to just randomly make suggestions.

**Reply by baszach** [reply]
It seems odd to me too but the former developer did this...

Thanks for the quick answers!
